tests/test_unions.py

```diff
@@ -0,0 +1,132 @@
+from unittest.mock import Mock
+
+import pytest
+from graphql import build_schema, graphql_sync
+
+from ariadne import ResolverMap, Union, make_executable_schema
+
+type_defs = """
+    union FeedItem = User | Thread
+
+    type User {
+        username: String
+    }
+
+    type Thread {
+        title: String
+    }
+
+    type Query {
+        item: FeedItem
+    }
+"""
+
+
+@pytest.fixture
+def schema():
+    return build_schema(type_defs)
+
+
+def test_attempt_bind_union_to_undefined_type_raises_error(schema):
+    union = Union("Test")
+    with pytest.raises(ValueError):
+        union.bind_to_schema(schema)
+
+
+def test_attempt_bind_union_to_invalid_type_raises_error(schema):
+    union = Union("Query")
+    with pytest.raises(ValueError):
+        union.bind_to_schema(schema)
+
+
+test_query = """
+    {
+        item {
+            __typename
+            ... on User {
+                username
+            }
+            ... on Thread {
+                title
+            }
+        }
+    }
+"""
+
+user = Mock(username="User")
```

Comment:
Naming module-scoped symbols should be different then local variables, especially when we have same naming convention for pytest fixtures.
I would make it _USER_ or _User_ at least. Same applies for thread.